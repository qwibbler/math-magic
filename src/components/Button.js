import React from 'react';
import PropTypes from 'prop-types';

export default class Button extends React.Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  render() {
    const { name } = this.props;
    return (
      <input type="button" id={name} value={name} />
    );
  }
}
Button.propTypes = {
  name: PropTypes.string.isRequired,
};
