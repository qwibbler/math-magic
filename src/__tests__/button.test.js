import React from 'react';
import renderer from 'react-test-renderer';
import Button from '../components/Button';

describe('Button', () => {
  test('Link renders correctly', () => {
    const tree = renderer
      .create(
        <Button
          name="test"
          click={function () {
            console.log('test');
          }}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
