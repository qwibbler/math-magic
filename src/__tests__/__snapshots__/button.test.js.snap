// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Link renders correctly 1`] = `
<input
  id="test"
  onClick={[Function]}
  type="button"
  value="test"
/>
`;
